### ISSUE TITLE
Build fails due to missing libxlt

### ISSUE THREAD

**Issue by seppd** [daphne-eu/daphne#692]
1. Take master @ [e7649ba](https://github.com/daphne-eu/daphne/commit/e7649ba6bdc629a03936e49c27ed876efbce6db4)
2. `git clone --recursive https://github.com/daphne-eu/daphne.git`
3. `./build.sh`

Build fails with the following error:
```
[1/6] Linking CXX shared library release/libarrow.so.1300.0.0
FAILED: release/libarrow.so.1300.0.0
...
/usr/bin/ld: cannot find -lxtl: No such file or directory
collect2: error: ld returned 1 exit status
```

**Comment by corepointer** [comment]
Hi!  I ran into this one some time ago. Iirc you need to get rid of xtensor packages installed in your system.

**Comment by seppd** [comment]
@corepointer Thanks for a hint. I don't think it's installed. Let me try to install it instead and see what happens.

**Comment by seppd** [comment]
No, it did not help. The related Ubuntu packages don't have a library, they contain headers (no wonder as xtl is a header-only library).

**Comment by seppd** [comment]
No, I don't:
```
apt search xtensor
Sorting... Done
Full Text Search... Done
libxtensor-blas-dev/mantic 0.20.0-2 amd64
  an extension to xtensor offering bindings to BLAS and LAPACK

libxtensor-dev/mantic 0.24.3-2ubuntu1 amd64
  multi-dimensional arrays with broadcasting and lazy computing

xtensor-blas-doc/mantic 0.20.0-2 all
  documentation for xtensor-blas

xtensor-dev/mantic 0.24.3-2ubuntu1 all
  multi-dimensional arrays with broadcasting (transition package)

xtensor-doc/mantic 0.24.3-2ubuntu1 all
  documentation for xtensor

xtl-dev/mantic 0.7.2-2.1 all
  basic tools (containers, algorithms) used for xtensor and xeus
```

Still fails with the same error after cleaning the Arrow's build directory. But this was expected as the initial error happened with a  completely fresh build after a clone.

**Comment by seppd** [comment]
The error can be avoided by providing `-DARROW_BUILD_SHARED=OFF`, but then later on daphne itself fails to link of course (`Arrow::arrow_shared` is missing)

**Comment by corepointer** [comment]
If you just wanna make it work, another workaround is to remove the ``-lxtl`` occurrences from the ninja build file in the build directory and starting the build and install steps from there manually.

**Comment by seppd** [comment]
Ok, based on this comment https://github.com/apache/arrow/issues/37146#issuecomment-1678308414 I found a workaround: remove the `libxsimd-dev` package. probably that was the one you meant @corepointer. 

Not sure if it can be considered as a solution, this kind of a trick might be fine within a container, but removing a package at the host system level does not  look good.
